tests/test_serialize.cc

```diff
@@ -211,156 +194,114 @@ struct DatasetWithMetadata
   };
 };
 
-class UnfitSerializableModel
-    : public SerializableType<SerializableMockPointer> {
-public:
-  RepresentationType create() const override {
-    return std::make_unique<MockModel>(log(2.));
-  }
-
-  bool are_equal(const RepresentationType &lhs,
-                 const RepresentationType &rhs) const override {
-    return *lhs == *rhs;
-  };
-};
+REGISTER_TYPED_TEST_CASE_P(SerializeTest, test_roundtrip_serialize_json,
```

Comment:
Neat! Didn't know gtest had this available.

Reply:
Yeah!  This works out quite nice.  I like how it lets you define a set of things you want tested, then you can end up using those in specialized tests (the way we're doing now with `ExampleModels`).  It's led me to some bugs that I would have been too lazy to have tested for otherwise.